<launch>
    <node name="steering_pid" pkg="pid" type="controller" output="screen" >
      <param name="node_name" value="steering_pid" />
      <param name="topic_from_controller" value="robot/pid/steering/control_effort" />
      <param name="topic_from_plant" value="robot/pid/steering/state" />
      <param name="setpoint_topic" value="robot/pid/steering/setpoint" />
      <param name="pid_enable_topic" value="robot/pid/steering/enable" />

      <param name="Kp" value="5.0" />
      <param name="Ki" value="0.0" />
      <param name="Kd" value="0.1" />
      <param name="upper_limit" value="1905" />
      <param name="lower_limit" value="-1905" />
      <param name="windup_limit" value="10" />
      <param name="cutoff_frequency" value="20" />
      <param name="max_loop_frequency" value="105.0" />
      <param name="min_loop_frequency" value="95.0" />
     </node>

     <node pkg="advanced_robotics_team6" type="ros_pid_test.py" name="pid_broadcast" output="screen" />

    <!--
    <node name="servo_sim_node" pkg="pid" type="plant_sim" output="screen" >
      <param name="plant_order" value="2" />
    </node> -->

    <!--
    <node name="setpoint_node" pkg="pid" type="setpoint_node" output="screen" >
      <param name="setpoint_topic" value="robot/pid/steering/setpoint" />
    </node>
  -->


    <!-- rqt_plot is a resource hog, so if you're seeing high CPU usage, don't launch rqt_plot  -->
    <!--
    <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"
    args="/control_effort/data /state/data /setpoint/data" />

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

    <node name="rqt_robot_monitor" pkg="rqt_robot_monitor" type="rqt_robot_monitor" />
  -->
</launch>
